src/generate-font-faces.js

```diff
@@ -1,29 +1,57 @@
-/** Copyright (c) 2018 Uber Technologies, Inc.
+/** Copyright (c) 2019 Uber Technologies, Inc.
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
  * @flow
  */
 
-export default function generateFontFaces(fontDictionary: {}) {
+import type {AtomicFontsObjectType, StyledFontsObjectType} from './types';
+
+export function generateAtomicFontFaces(fonts: AtomicFontsObjectType) {
   const faces = [];
-  Object.keys(fontDictionary).forEach(fontName => {
-    const font = fontDictionary[fontName];
+  Object.keys(fonts).forEach(fontName => {
+    const font = fonts[fontName];
     if (font) {
       faces.push(
-        `@font-face {font-family: "${fontName}"; font-display: fallback; src: ${String(
-          asFontFaceSrc(font.urls)
-        )};}`
+        `@font-face {
+          font-family: "${fontName}";
+          font-display: fallback;
+          src: ${String(asFontFaceSrc(font.urls))};
+          ${String(asFontFaceStyles(font.styles || {}))}
+        }`
       );
     }
   });
   return '\n' + faces.join('\n');
 }
 
+export function generateStyledFontFaces(fonts: StyledFontsObjectType) {
+  const faces = [];
+  Object.keys(fonts).forEach(fontName => {
+    fonts[fontName].forEach(fontInstance => {
+      faces.push(
+        `@font-face {
+font-family: "${fontName}";
+font-display: fallback;
+src: ${asFontFaceSrc(fontInstance.urls).join(',\n')};
+${String(asFontFaceStyles(fontInstance.styles))}}`
+      );
+    });
+  });
+  return '\n' + faces.join('\n');
+}
+
 function asFontFaceSrc(urls) {
   // `urls` is a dictionary of font types (woff, woff2 etc) to url string
   return Object.keys(urls).map(
-    type => `url("${urls[type]}") format("${type}")\n`
+    // $FlowFixMe
```

Comment:
Hm, curious as to why this is necessary?

Reply:
I'm not entirely sure but it relates to
```ts
type FontURLsType = {
  woff?: string,
  woff2: string,
};
```

Because `woff` is an optional key it is somehow inferring that `type` might be undefined.

Reply:
Ah, that's annoying.  I think this is because an object with the propery `woff: undefined` satisfies the type constraints _and_ would fail the `urls[type]` coercion because it may be `undefined`.  Something like this:

```js
const urls: FontURLsType = {
  woff: undefined,
  woff2: "hello"
};

console.log(Object.keys(urls)); // ['woff', 'woff2'];
```

If you want to get rid of this `$FlowFixMe` you could wrap the offending code in a `String` constructor:

```
type => `url("${String(urls[type])}") format("${type}")`
```

which should resolve this.

Reply:
👍